src/Core/Validation/FieldValidation/FieldValidationTrait.php

```diff
@@ -61,12 +73,10 @@ private function getFieldValidators(): array
             throw new RuntimeException($message);
         }
         /** @var FieldValidationInterface|Configurable $this */
-        $name = $this->getName();
+        // Use a default name if the field doesn't have one rather than throwing an exception
+        // because some fields may not have a name, i.e. CompositeField
+        $name = $this->getName() ?: _t(__CLASS__ . '.UNNAMEDFIELD', 'Unnamed field');
```

Comment:
I can't remember the scenario, though I think this is out of scope given this about validation rather than the name that CompositeField has.  Subclasses of CompositeField could also just override whatever fix we put in. I think it's better to keep this as they are as this code is robust against whatever a field does with its name.